### ISSUE TITLE
Recursively get property downwards for `Add` and `Mul`

### ISSUE THREAD

**Issue by icarosadero** [sympy/sympy#26538]
Some objects, like the ones from the `physics.quantum` submodule, contain properties like that yield a derived expression. However, this same property can't be accessed for a linear combination of these same objects.

For example,

Suppose I create two `Ket` objects:

```python
from sympy.physics.quantum.state import Ket
x = Ket('1')
y = Ket('2')
```

Then, I could call the property `dual` separately on kets $|1\rangle$ and $|2\rangle$ without a problem to generate bras from them:

```python
In: x.dual
Out: <1|

In: y.dual
Out: <2|
```

However, for any linear combination of $|1\rangle$ and $|2\rangle$, that no longer works:

```python
In: (2*x + 3*y).dual
```

```bash
AttributeError                            Traceback (most recent call last)
Cell In[9], line 1
----> 1 (2*x + 3*y).dual

AttributeError: 'Add' object has no attribute 'dual'
```

This forces the user to implement a recursive function to call `dual` on the expression on every new project. This also applies to `expr` from `physics.quantum.state.Wavefunction`, for which I created a PR #26520 to temporarily solve it. 

The solution that I suggest is to "send" a property or method call down the expression tree whenever it is invalid for `Add` or `Mul`.

**Comment by sylee957** [comment]
I don't think that the discussion ends with Add and Mul, not only Add don't have `dual`, but also most of the SymPy methods like `Integer`, `Rational`. I'm not very clear about how are you going to handle more complicated cases.

Also the programming practice like 

> "send" a property or method call down the expression tree whenever it is invalid for Add or Mul.

this can be very complicated and easily become counter intuitive for users, and against pythonic practices.

**Comment by ellisonbg** [comment]
Implementing this would be quite difficult, as it would inject methods/attrs from arbitrary Expr subclasses into things like Add/Mul/Pow. However, you can do what you want using `Dagger`:

```python
Dagger(2*x + 3*y)
```
